File: services/shop-backend/src/main/java/io/miragon/shop/application/port/outbound/OrderRepository.java
package io.miragon.shop.application.port.outbound;

import io.miragon.shop.domain.order.Order;
import io.miragon.shop.domain.order.OrderId;
import io.miragon.shop.domain.shared.UserId;

import java.util.List;

public interface OrderRepository {
    Order save(Order order);
    Order loadById(OrderId orderId);
    List<Order> loadById(UserId userId);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/application/port/outbound/OrderRepository.kt
package io.miragon.shop.application.port.outbound

import io.miragon.shop.domain.order.Order
import io.miragon.shop.domain.order.OrderId
import io.miragon.shop.domain.shared.UserId

interface OrderRepository {
    fun save(order: Order): Order
    fun loadById(orderId: OrderId): Order?
    fun loadById(userId: UserId): List<Order>
}